examples/with-react-hook-form/src/app/page.tsx

```diff
@@ -0,0 +1,73 @@
+"use client";
+
+import { useState } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+
+import { useDropzone } from "@uploadthing/react/hooks";
+
+import { useUploadThing } from "~/utils/uploadthing";
+
+type Inputs = {
+  message: string;
+};
+
+export default function Home() {
+  const [files, setFiles] = useState<File[]>([]);
+  const { startUpload } = useUploadThing("videoAndImage");
+  const { handleSubmit, register, reset } = useForm<Inputs>();
+  const { getInputProps, getRootProps } = useDropzone({
+    onDrop: (files) => {
+      setFiles(files);
+    },
+  });
+
+  const onSubmit: SubmitHandler<Inputs> = async (data) => {
+    if (!files) return;
+
+    const res = await startUpload(files);
+
+    if (!res) {
+      alert("Upload failed");
+      return;
+    }
+
```

Comment:
I'd maybe leave some comments in here explaining what information we have at different points (eg. file itself, uploaded url, other form data) and maybe a mock of how a user might want to store info in their db.